components/service/experiments/src/test/java/mozilla/components/service/experiments/ExperimentEvaluatorTest.kt

```diff
@@ -642,4 +502,140 @@ class ExperimentEvaluatorTest {
                     Assert.assertTrue(it.value in 350..650)
                 }
     }
+
+    @Test
+    fun `test selectBranchByWeight()`() {
+        val e = ExperimentEvaluator()
+
+        var branches = listOf(
+            Experiment.Branch("1", 2)
+        )
+
+        assertEquals("1", e.selectBranchByWeight(0, branches).name)
+        assertEquals("1", e.selectBranchByWeight(1, branches).name)
+
+        branches = listOf(
+            Experiment.Branch("1", 1),
+            Experiment.Branch("2", 2)
+        )
+
+        assertEquals("1", e.selectBranchByWeight(0, branches).name)
+        assertEquals("2", e.selectBranchByWeight(1, branches).name)
+        assertEquals("2", e.selectBranchByWeight(2, branches).name)
+
+        branches = listOf(
+            Experiment.Branch("1", 2),
+            Experiment.Branch("2", 4),
+            Experiment.Branch("3", 1)
+        )
+
+        assertEquals("1", e.selectBranchByWeight(0, branches).name)
+        assertEquals("1", e.selectBranchByWeight(1, branches).name)
+        assertEquals("2", e.selectBranchByWeight(2, branches).name)
+        assertEquals("2", e.selectBranchByWeight(3, branches).name)
+        assertEquals("2", e.selectBranchByWeight(4, branches).name)
+        assertEquals("2", e.selectBranchByWeight(5, branches).name)
+        assertEquals("3", e.selectBranchByWeight(6, branches).name)
+    }
+
+    @Test(expected = AssertionError::class)
+    fun `test selectBranchByWeight() asserts upper out-of-bounds`() {
+        val branches = listOf(
+            Experiment.Branch("1", 2)
+        )
+        val evaluator = ExperimentEvaluator()
+        evaluator.selectBranchByWeight(2, branches)
+    }
+
+    @Test(expected = AssertionError::class)
+    fun `test selectBranchByWeight() asserts lower out-of-bounds`() {
+        val branches = listOf(
+            Experiment.Branch("1", 2)
+        )
+        val evaluator = ExperimentEvaluator()
+        evaluator.selectBranchByWeight(2, branches)
+    }
+
+    @Test
+    fun `test pickActiveBranch()`() {
+        val e = ExperimentEvaluator()
+
+        var experiment = testDefaultExperiment(
+            branches = listOf(
+                Experiment.Branch("1", 2)
+            )
+        )
+
+        assertEquals("1", e.pickActiveBranch(experiment, { _, _ -> 0 }).name)
+        assertEquals("1", e.pickActiveBranch(experiment, { _, _ -> 1 }).name)
```

Comment:
nit: the IDE linter says you can move the lambda out of the parentheses, so this line could be:
```Kotlin
assertEquals("1", e.pickActiveBranch(experiment) { _, _ -> 1 }.name)
```